test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -153,6 +177,139 @@ public void Controller_View_WithParameterViewNameAndViewModel_SetsResultViewName
+        }
+
+        [Fact]
+        public void Controller_PartialView_WithParameterViewName_SetsResultViewNameAndNullViewDataModelAndSameTempData()
+        {
+            // Arrange
+            var controller = new TestableController()
+            {
+                ViewData = new ViewDataDictionary(new EmptyModelMetadataProvider()),
+                TempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>()),
+            };
+
+            // Act
+            var actualViewResult = controller.PartialView("CustomViewName");
+
+            // Assert
+            Assert.IsType<PartialViewResult>(actualViewResult);
+            Assert.Equal("CustomViewName", actualViewResult.ViewName);
+            Assert.Same(controller.ViewData, actualViewResult.ViewData);
+            Assert.Same(controller.TempData, actualViewResult.TempData);
+            Assert.Null(actualViewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Controller_PartialView_WithParameterViewModel_SetsResultNullViewNameAndViewDataModelAndSameTempData()
+        {
+            // Arrange
+            var controller = new TestableController()
+            {
+                ViewData = new ViewDataDictionary(new EmptyModelMetadataProvider()),
+                TempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>()),
+            };
+            var model = new object();
+
+            // Act
+            var actualViewResult = controller.PartialView(model);
+
+            // Assert
+            Assert.IsType<PartialViewResult>(actualViewResult);
+            Assert.Null(actualViewResult.ViewName);
+            Assert.Same(controller.ViewData, actualViewResult.ViewData);
+            Assert.Same(controller.TempData, actualViewResult.TempData);
+            Assert.Same(model, actualViewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Controller_PartialView_WithParameterViewNameAndViewModel_SetsResultViewNameAndViewDataModelAndSameTempData()
+        {
+            // Arrange
+            var controller = new TestableController()
+            {
+                ViewData = new ViewDataDictionary(new EmptyModelMetadataProvider()),
+                TempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>()),
+            };
+            var model = new object();
+
+            // Act
+            var actualViewResult = controller.PartialView("CustomViewName", model);
+
+            // Assert
+            Assert.IsType<PartialViewResult>(actualViewResult);
+            Assert.Equal("CustomViewName", actualViewResult.ViewName);
+            Assert.Same(controller.ViewData, actualViewResult.ViewData);
+            Assert.Same(controller.TempData, actualViewResult.TempData);
+            Assert.Same(model, actualViewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Controller_PartialView_WithNullModelParameter_MaintainsNullModel()
```

Comment:
Change name to reflect fact the `Model` is _changed_ (set) from non-`null` to `null` in this scenario.